nix-language: attempt to simplify concept overview (#397)
NixOS/nix.dev: source/tutorials/nix-language.md

The Nix language has only few basic constructs which can be combined arbitrarily:

> - Primitive data types i.e. strings, integers, booleans
> - Compound data types i.e lists and attribute sets
> - Functions to produce and transform data
> - Name assignment to manipulate data as units

Reviewer: ```suggestion
- Primitive data types

  such as integers or character strings

- Compound data types

  that is, lists and attribute sets

- Functions and operators

  to produce and transform data

- Name assignment

  to manipulate data as units
```
